db/user.go

```diff
@@ -0,0 +1,81 @@
+package db
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+var ErrNoUser = errors.New("no user")
+
+type User struct {
+	Id    int    `json:"id"`
+	Email string `json:"email"`
+	Token string `json:"token"`
+}
+
+func FindUserBy(column string, value interface{}) (*User, error) {
+	u := &User{}
+	q := fmt.Sprintf("SELECT id, email, token FROM users WHERE %v=$1", column)
+
+	if err := db.Get(u, q, value); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ErrNoUser
+		}
+		return nil, err
+	}
+	return u, nil
+}
+
+func FindUserByEmail(email string) (*User, error) {
+	return FindUserBy("email", email)
+}
+
+func FindUserById(id int) (*User, error) {
+	return FindUserBy("id", id)
+}
+
+func FindUserByToken(token string) (*User, error) {
+	return FindUserBy("token", token)
+}
+
+func CreateUser(email string) (*User, error) {
+	token, err := generateUniqueToken()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.Exec(`INSERT INTO users (email, token) VALUES ($1, $2)`, email, token)
+	if err != nil {
+		return nil, err
```

Comment:
Can we get a better idea of what the error is and respond with an appropriate message?